### PR TITLE
Try next dep. resolver if building env failed

### PR DESCRIPTION
**Pull request by mvdbeek**
Otherwise an exception ("Conda dependency seemingly installed but failed to
build job environment." comes to mind) would entirely stop the job submission.
Ping @bgruening.

### CODE REVIEW

**Comment by mvdbeek** [comment]
Two factors are potentially increasing the frequency with which we can see this error: Conda versions newer than 4.3.24 also have problems with `--offline`, and bcftools (in addition to samtools) being a dependency to "Set external metadata", which requires a mulled environment. The entirely correct thing on the deployer side would be to install these dependencies through the admin panel, but upgrading an instance to find that even simple uploads don't work is not great UX.

File: lib/galaxy/tools/deps/__init__.py [hunk 1]
```diff
@@ -111,38 +111,43 @@ def get_app_option(self, key, default=None):
...
+            ordered_dependencies = OrderedSet(dependencies.values())
+            try:
+                shell_commands = [dependency.shell_commands() for dependency in ordered_dependencies]
+                if 'tool_instance' in kwds:
+                    # We log the dependencies on the tool instance,
+                    # which subsequently is used to log the used dependencies for this job.
+                    kwds['tool_instance'].dependencies = [dep.to_dict() for dep in ordered_dependencies]
+            except Exception as e:
+                log.exception(e)
+                pass
```
Comment (bgruening) [inline lib/galaxy/tools/deps/__init__.py]: the `pass` can be removed isn't it?

**Review by bgruening — COMMENTED** [review]
LGTM, thanks @mvdbeek!

File: lib/galaxy/tools/deps/__init__.py (hunk 1 above)
Comment (mvdbeek) [reply lib/galaxy/tools/deps/__init__.py]: right

**Comment by jmchilton** [comment]
Awesome - thanks @mvdbeek !

**Pull request merged**